src/shared/vscode/window.ts

```diff
@@ -0,0 +1,80 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+
+/**
+ * Components associated with {@link vscode.window}.
+ */
+export interface Window {
+    statusBar: StatusBar
+    inputBox: InputBox
+    message: Message
+    progress: Progress
+    dialog: Dialog
+}
+
+/**
+ * Actions associated with status bars in {@link vscode.window}.
+ */
+export interface StatusBar {
+    /**
+     * See {@link vscode.window.setStatusBarMessage}.
+     */
+    setMessage(message: string, hideAfterTimeout?: number): vscode.Disposable
+}
+
+/**
+ * Actions associated with input boxes in {@link vscode.window}.
+ */
+export interface InputBox {
+    /**
+     * See {@link vscode.window.showInputBox}.
+     */
+    show(options?: vscode.InputBoxOptions, token?: vscode.CancellationToken): Thenable<string | undefined>
+}
+
+/**
+ * Actions associated with showing messages in {@link vscode.window}.
+ */
+export interface Message {
+    /**
+     * See {@link vscode.window.showErrorMessage}.
+     */
+    showError(message: string, ...items: string[]): Thenable<string | undefined>
```

Comment:
ShowXxxMessage are good examples in the vscode api where there are a bunch of overloads (see vscode's index.d.ts file). These are gross to work with at the mocking level. Do you have a plan to minimize pain in this space?

(an example use-case is to call a modal ShowXXXMessage )

Reply:
There are indeed a lot of overloads and variations of a lot of these. Luckily I think we're only using a small subset of them for normal cases. I think we can implement them as they come up. 

Branches in the logic caused by different parameters do make the mocking more difficult. However, most of these implementations simply capture the arguments and return a stub response. So dealing with variations in parameters shouldn't be nearly as difficult as in the implementation of the real vscode apis.